### ISSUE TITLE
number, time and count functions type mismatch

### ISSUE THREAD

**Issue by Fxlr8** [nanostores/i18n#10]
```ts
const { number, time } = useStore(format)
```

The documentation clearly says that they accept a second argument:
https://github.com/nanostores/i18n/blob/99d21a5b709db9a1127b2ac529b3765f9be1dce6/README.md?plain=1#L209-L226

But your typings do not agree with that:
<img width="631" alt="image" src="https://user-images.githubusercontent.com/10725007/151545538-6d75dd10-7daf-4b16-aaf9-fc31759c8676.png">

**Comment by Fxlr8** [comment]
Also the `count` function type says that it returns a function:
<img width="507" alt="image" src="https://user-images.githubusercontent.com/10725007/151545876-b9065560-2b4b-4c43-9765-28f1797f2590.png">
But if you try to call it, you get this
```
Uncaught TypeError: formatBanana is not a function
```

**Comment by ai** [comment]
It happens because of the error in TypeScript `Intl.NumberFormat.format` types.

We do not create our own types, but re-use TS types for `Intl` module https://github.com/nanostores/i18n/blob/main/formatter/index.d.ts#L6-L7

Seems like we need to migrate to custom types.

**Comment by ai** [comment]
Can you create a separate issue for `count`?

**Comment by ai** [comment]
Fixed 2a9cae3

**Comment by ai** [comment]
> Also the count function type says that it returns a function

`count` should be used only inside `i18n({ … })` call.

TypeScript dynamic types can be tricky. I didn’t found a way to solve type problem of converting types inside `i18n` to `t` types. Think about it as trade off for types maintainability.

https://github.com/nanostores/i18n/blob/main/create-i18n/index.d.ts#L36-L44

**Comment by ai** [comment]
Released in 0.2.1.
